examples/http-server-using-redis/main.go

```diff
@@ -71,3 +76,31 @@ func RedisPipelineHandler(c *gofr.Context) (interface{}, error) {
 	// Process or return the results of each command in the pipeline (implementation omitted for brevity)
 	return cmds, nil
 }
+
+// RedisLockHandler demonstrates how to acquire a Redis lock, perform a simple increment operation under
+// the lock, and ensure the lock is released afterward.
+func RedisLockHandler(c *gofr.Context) (interface{}, error) {
+	// Try to obtain the lock
+	lock, err := c.Redis.Locker().Obtain(c, "my-lock-key", 5*time.Second, nil)
+	if err != nil || errors.Is(err, redislock.ErrNotObtained) {
+		return nil, err
+	}
+
+	// If lock is successfully acquired, ensure it is released when the handler exits
+	defer lock.Release(c)
+
+	// Perform a meaningful operation under the lock
+	countKey := "locked-operation-count"
+	count, err := c.Redis.Get(c, countKey).Int()
+	if err != nil && !errors.Is(err, redis.Nil) {
+		return nil, err
+	}
+
+	// Increment the count under the lock
+	err = c.Redis.Set(c, countKey, count+1, redisExpiryTime*time.Minute).Err()
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
I am unsure if this would work
Because in the example we are acquiring a lock and then using c.Redis to get and set the values.

So, if two parallel requests come in - both of them will be processed without wait inspite of having the lock.

If this would have been container/pod level maybe this would have worked - but it should not be like that.

or are we acquiring the lock only on the my-lock-key - if so how it is being demonstrated here?